build.go

```diff
@@ -0,0 +1,355 @@
+package composer
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/paketo-buildpacks/packit/v2"
+	"github.com/paketo-buildpacks/packit/v2/draft"
+	"github.com/paketo-buildpacks/packit/v2/fs"
+	"github.com/paketo-buildpacks/packit/v2/pexec"
+	"github.com/paketo-buildpacks/packit/v2/scribe"
+)
+
+// DetermineComposerInstallOptions defines the interface to get options for `composer install`
+//go:generate faux --interface DetermineComposerInstallOptions --output fakes/determine_composer_install_options.go
+type DetermineComposerInstallOptions interface {
+	Determine() []string
+}
+
+// Executable just provides a fake for pexec.Executable for testing
+//go:generate faux --interface Executable --output fakes/executable.go
+type Executable interface {
+	Execute(pexec.Execution) (err error)
+}
+
+// Calculator defines the interface for calculating a checksum of the given set
+// of file paths.
+//go:generate faux --interface Calculator --output fakes/calculator.go
+type Calculator interface {
+	Sum(paths ...string) (string, error)
+}
+
+func Build(
+	logger scribe.Emitter,
+	composerInstallOptions DetermineComposerInstallOptions,
+	composerInstallExec Executable,
+	composerGlobalExec Executable,
+	checkPlatformReqsExec Executable,
+	path string,
+	calculator Calculator) packit.BuildFunc {
+	return func(context packit.BuildContext) (packit.BuildResult, error) {
+		logger.Title("%s %s", context.BuildpackInfo.Name, context.BuildpackInfo.Version)
+
+		composerPhpIniPath, err := writeComposerPhpIni(logger, context)
+		if err != nil {
+			return packit.BuildResult{}, err
+		}
+
+		composerGlobalBin, err := runComposerGlobalIfRequired(logger, context, composerGlobalExec, path, composerPhpIniPath)
+		if err != nil {
+			return packit.BuildResult{}, err
+		}
+
+		if composerGlobalBin != "" {
+			path = strings.Join([]string{
+				composerGlobalBin,
+				path,
+			}, string(os.PathListSeparator))
+		}
+
+		workspaceVendorDir := filepath.Join(context.WorkingDir, "vendor")
+
+		if value, found := os.LookupEnv(ComposerVendorDir); found {
+			workspaceVendorDir = filepath.Join(context.WorkingDir, value)
+		}
+
+		composerPackagesLayer, layerVendorDir, err := runComposerInstall(
+			logger,
+			context,
+			composerInstallOptions,
+			composerPhpIniPath,
+			path,
+			composerInstallExec,
+			workspaceVendorDir,
+			calculator)
+		if err != nil {
+			return packit.BuildResult{}, err
+		}
+
+		logger.Process("Writing symlink to %s", workspaceVendorDir)
+
+		err = os.Symlink(layerVendorDir, workspaceVendorDir)
+		if err != nil {
+			return packit.BuildResult{}, err
+		}
+
+		if os.Getenv(BpLogLevel) == "DEBUG" {
+			logger.Debug.Action("listing files in %s:", layerVendorDir)
+			files, err := ioutil.ReadDir(layerVendorDir)
+			if err != nil {
+				return packit.BuildResult{}, err
+			}
+			for _, f := range files {
+				logger.Debug.Detail(fmt.Sprintf("- %s", f.Name()))
+			}
+		}
+
```

Comment:
Same as other comment, why do we need to check for `DEBUG` being set?

Reply:
The purpose was to avoid reading from the filesystem if the logs would be ignored anyways. It's common to see in Java logging setups but a bit awkward here. I'll see if there's A Better Way ©.